l10n_pt_account_invoicexpress/models/account_move.py

```diff
@@ -119,12 +120,23 @@ def _prepare_invoicexpress_lines(self):
             tax = line.tax_ids[:1]
             # If not tax set, force zero VAT
             tax_detail = {"name": tax.name or "IVA0", "value": tax.amount or 0.0}
+            # Because InvoiceXpress expects unit_price in EUR, check if we need to convert
+            # line currency to company currency (company should use EUR as default currency)
+            if line.currency_id == line.company_id.currency_id:
+                price_unit = line.price_unit
+            else:
+                price_unit = line.currency_id._convert(
+                    line.price_unit,
+                    line.company_id.currency_id,
+                    line.company_id,
+                    date_today,
```

Comment:
Why today, shouldn't it be the Invoice Date?
What date is used to convert the amount used for the accounting entries?

Reply:
@dreispt invoice date seems more reasonable. I checked the way Odoo is calculating the currency rate for the invoice line and did the same.